components/concept/engine/src/main/java/mozilla/components/concept/engine/Engine.kt

```diff
@@ -221,6 +222,24 @@ interface Engine : WebExtensionRuntime, DataCleanable {
         serviceWorkerDelegate: ServiceWorkerDelegate
     ): Unit = throw UnsupportedOperationException("Service workers support not available in this engine")
 
+    /**
+     * Un-registers the attached [ServiceWorkerDelegate] if one was added with
+     * [registerServiceWorkerDelegate].
+     */
+    fun unregisterServiceWorkerDelegate(): Unit =
+        throw UnsupportedOperationException("Service workers support not available in this engine")
+
+    /**
+     * Handle user interacting with a web notification.
+     *
+     * @param webNotification [Parcelable] representing a web notification.
+     * This is not guaranteed to be valid. Receivers should check to ensure validity.
```

Comment:
^^ What do you mean by this (not guaranteed to be valid)? Do you men that the Parcelable doesn't necessarily need to be a `WebNotification`? If so I would just describe what happens if it's not e.g., nothing :). Because we use the safe cast operator.